### PR TITLE
bump stack all versions

### PR DESCRIPTION
**Pull request by jappeace**
This modifies the stack install script to be installed on all architectures (although it still likely will fail on sha, but I intend to get those from the error message).
The stack install script has also been moved outside of the docker file for easier future edits. (hence the large amount of deletions).

this also adds support for debian aarch64, solving https://github.com/haskell/docker-haskell/issues/59

### CODE REVIEW

File: install-stack.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+set -eux;
+
+cd /tmp;
+ARCH="$(dpkg-architecture --query DEB_BUILD_GNU_CPU)";
+STACK_URL="https://github.com/commercialhaskell/stack/releases/download/v${STACK}/stack-${STACK}-linux-$ARCH.tar.gz";
+# sha256 from https://github.com/commercialhaskell/stack/releases/download/v${STACK}/stack-${STACK}-linux-$ARCH.tar.gz.sha256
+case "$ARCH" in
+    'aarch64')
+        STACK_SHA256='0581cebe880b8ed47556ee73d8bbb9d602b5b82e38f89f6aa53acaec37e7760d';
+        ;;
```
Comment (eruizc-dev) [inline install-stack.sh:12]: This should be mentioned somewhere, let it be the title or the description

**Review by eruizc-dev — APPROVED** [review]

File: install-stack.sh (hunk 1 above)
Comment (jappeace) [reply install-stack.sh:12]: thanks, this sha is a lie, it'll likely break the script, I need ci to tell me what this is.

**Comment by AlistairB** [comment]
Thanks!

My suggestion would be to not rewrite how the images work until you understand and are comfortable with the current pattern. For example, the directory structure exists as that is how the official images [reference](https://github.com/docker-library/official-images/blob/master/library/haskell#L7) the dockerfiles for different GHC versions.

In terms of separate install scripts.. I don't know why not.. but this is not really done in any of the official images. The current install pattern pretty much[ mirrors all the official images](https://github.com/docker-library/golang/blob/master/1.19/buster/Dockerfile#L25-L120). I suppose for one you are forced to have an image layer with the install script sticking around which is not valuable to the end user. You could [create an issue on the official images repo](https://github.com/docker-library/official-images/issues) asking them about the pros and cons of the install script being copied in solution.

If you want to fix the duplication problem, I think like the go official images the right solution is to use [templating](https://github.com/haskell/docker-haskell/issues/65). 
In general, we should following the patterns from the other official images which have been developed by very knowledge people, unless we have a good reason.

In terms of the shas, the dockerfiles have a comments like `# sha256 from https://github.com/commercialhaskell/stack/releases/download/v${STACK}/stack-${STACK}-linux-$ARCH.tar.gz.sha256` telling you where to get them. Still, this is clearly not obvious enough. I'll spend some time on the weekend making sure this info is clear in the README.

So yeah, I suggest just doing the minimum to bump stack for now without other changes. If you want to do bigger changes that could go in a separate PR. Or feel free to create an issue.

**Comment by AlistairB** [comment]
Oh and the [GPG release key for stack](https://docs.haskellstack.org/en/stable/SIGNING_KEY/) never changes. I'll try and incorporate this better into the docs.

**Comment by AlistairB** [comment]
I have gone ahead and done this bump. See https://github.com/haskell/docker-haskell/issues/72#issuecomment-1365508001 for details.

**Pull request closed without merging**